pkg/cgroup/cgroup.go

```diff
@@ -119,12 +73,14 @@ func PathV2AddMountpoint(path string) (string, error) {
 
 // ID returns the cgroup2 ID of a path.
 func ID(pathWithMountpoint string) (uint64, error) {
-	cPathWithMountpoint := C.CString(pathWithMountpoint)
-	ret := uint64(C.get_cgroupid(cPathWithMountpoint))
-	C.free(unsafe.Pointer(cPathWithMountpoint))
-	if ret == 0 {
-		return 0, fmt.Errorf("GetCgroupID on %q failed", pathWithMountpoint)
+	hf, _, err := unix.NameToHandleAt(unix.AT_FDCWD, pathWithMountpoint, 0)
+	if err != nil {
+		return 0, fmt.Errorf("GetCgroupID on %q failed: %w", pathWithMountpoint, err)
+	}
+	if hf.Size() != 8 {
+		return 0, fmt.Errorf("GetCgroupID on %q failed: unexpected size", pathWithMountpoint)
 	}
+	ret := *(*uint64)(unsafe.Pointer(&hf.Bytes()[0]))
```

Comment:
Writing down some notes, as I was confused at first. Reading [the source code](https://cs.opensource.google/go/x/sys/+/master:unix/syscall_linux.go;l=2270;drc=a6f4650873b0647f4a7562d1bc45c27d4da686d7;bpv=1;bpt=1?q=NameToHandleAt&ss=go%2Fx%2Fsys) helped quite a bit 😄 .

Please feel free to correct this if I got some stuff wrong.

#### Before

We always allocated `cgid_file_handle` with `uint64_t cgid` (the handle data), which should always fit.

#### Now

[`.Bytes` + `.Type` are allocated + 32 Bytes for the handle](https://cs.opensource.google/go/x/sys/+/master:unix/syscall_linux.go;l=2279;drc=a6f4650873b0647f4a7562d1bc45c27d4da686d7;bpv=1;bpt=1?q=NameToHandleAt&ss=go%2Fx%2Fsys), which should be enough for us.

Then the `Bytes()` call that was throwing me off because thought that it was returning the whole raw buffer, including `Bytes` and `Type`, but [they are skipped](https://cs.opensource.google/go/x/sys/+/master:unix/syscall_linux.go;l=2270;drc=a6f4650873b0647f4a7562d1bc45c27d4da686d7;bpv=1;bpt=1?q=NameToHandleAt&ss=go%2Fx%2Fsys).


One more thing I wonder why they did `.Bytes +4` rather than something like the below which I think would be easier for maintenance and clearer? I guess it's due to perf reasons, but not sure.

```golang
return unsafe.Slice((*byte(unsafe.Pointer(uintptr(unsafe.Pointer(&fh.fileHandle.Bytes))+unsafe.Sizeof(&fh))), n)
```